mangaki/mangaki/views.py

```diff
@@ -749,6 +751,29 @@ def update_newsletter(request):
     return HttpResponse()
 
 
+def update_research(request):
+    is_ok = None
+    if request.user.is_authenticated and request.method == 'POST':  # Toggle on one's profile
+        username = request.user.username
+        is_ok = request.POST['research_ok'] == 'true'
+        Profile.objects.filter(user__username=username).update(research_ok=is_ok)
+        return HttpResponse()
+    if request.method == 'POST':  # Confirmed from mail
+        is_ok = 'yes' in request.POST
+        username = request.POST['username']
+        token = request.POST['token']
+    elif request.method == 'GET':  # Clicked on mail link
+        username = request.GET['username']
+        token = request.GET['token']
+    message = settings.HASH_NACL + username
```

Comment:
You are right, it was [not a good idea](https://en.wikipedia.org/wiki/Hash-based_message_authentication_code). But timestamp is not the good way to go. Thank you for pointing that out!